x-pack/plugins/enterprise_search/public/applications/shared/hidden_text/hidden_text.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, ReactElement } from 'react';
+
+interface ChildrenProps {
+  toggle: () => void;
+  isHidden: boolean;
+  hiddenText: string;
+}
+
+interface Props {
+  text: string;
+  children(props: ChildrenProps): ReactElement;
+}
+
+export const HiddenText: React.FC<Props> = ({ text, children }) => {
+  const [isHidden, toggleIsHidden] = useState(true);
+  const hiddenText = isHidden ? text.replace(/./g, '*') : text;
```

Comment:
This is 100% my obsessive design roots showing, but any chance we can use ••• symbols instead of *** due to vertical alignment? The first thing I noticed in the screenshots was that the stars weren't aligned with the copy icons, and thought that was a flexbox issue, but then I realized *s are just naturally high-aligned 🤦‍♀️ 

<img width="1126" alt="" src="https://user-images.githubusercontent.com/549407/95528453-055dbc80-098d-11eb-9f20-32719129a997.png">

```suggestion
  const hiddenText = isHidden ? text.replace(/./g, '•') : text;
```

What do you think? Yay/nay?

Reply:
ab4511344e43f6905d5142b1af96f0370d9c6cc0

Reply:
b231e8c6540c7b983d6ae1ac779eb93ee4c89347